api/src/services/application-csv-export.service.ts

```diff
@@ -122,80 +124,109 @@ export class ApplicationCsvExporterService
         .on('close', () => {
           resolve();
         })
-        .on('open', () => {
+        .on('open', async () => {
           writableStream.write(
             csvHeaders
               .map((header) => `"${header.label.replace(/"/g, `""`)}"`)
               .join(',') + '\n',
           );
 
-          // now loop over applications and write them to file
-          applications.forEach((app) => {
-            let row = '';
-            let preferences: ApplicationMultiselectQuestion[];
-            csvHeaders.forEach((header, index) => {
-              let multiselectQuestionValue = false;
-              let parsePreference = false;
-              let value = header.path.split('.').reduce((acc, curr) => {
-                // return preference/program as value for the format function to accept
-                if (multiselectQuestionValue) {
-                  return acc;
-                }
+          for (let i = 0; i < applications.length / 1000 + 1; i++) {
+            // grab applications 1k at a time
```

Comment:
How might this work with lotteries when we will have to give every application a random rank, and also have separate tabs by claimed preference and give those a rank?

Reply:
good questions! so up above this we do still grab all the application ids that will eventually find their way into the export. We do this still to grab the data of each application that'll show up in the export. So we can use that first query to do the ranking and possibly ordering the preferences

hopefully that makes some sense but lemme know if not